### DISCUSSION TITLE
Is it safe to make the backend container run in normal http, if the nginx-proxy already has ssl configured?

### DISCUSSION THREAD

**Discussion by nicholasyin** [nginx-proxy/nginx-proxy#2032] [Q&A] [answered]
I've put the ssl certificate's nginx version on the ubuntu host, done the volume mounting, and configured nginx-proxy to find certificates from /etc/nginx/certs. And it works fine now. From outside I can access using https and it looks like the backend service is serving https. But based on my understanding, in this case, the security work is done on the nginx-proxy. My backend container is running in normal http, listening on port 80. So can I safely leave it like this and not further configure to make the backend container run with https as well?

**Comment by tkw1536** [comment, marked as answer]
This depends on your definition of "safe".

Communication roughly works as follows:

Client <---> Docker Host nginx-proxy container <---> Application container

The first link, between the client and the docker host is encrypted with https. 
As you said, the second link uses http.
This could be changed to https using [`VIRTUAL_PROTO`](https://github.com/nginx-proxy/nginx-proxy#ssl-backends).

Data Packets on the first link can in theory be intercepted by a third party, like a malicous ISP sitting between your client and your server.
As it is encrypted with https, this should not be an issue.

Data Packets on the second link do not leave the docker host.
As such it is not possible for third parties to intercept any data on them. 

If however there is an attacker that gains access to your server, they could intercept data on the second link as well.
But in such a scenario where an attacker has control over your server, they likely could just control the application directly, or access your database.
